src/pages/gear.page.tsx

```diff
@@ -1,11 +1,85 @@
+import { useEffect, useState } from 'react'
+import { ParchmentInput } from '../components/ParchmentInput'
 import { PageHeader } from '../components/page-header'
 import { Parchment } from '../components/parchment'
 import { rollD6 } from '../functions/dice.functions'
+import { ParchmentButton } from '../components/ParchmentButton'
+import { ArrowPathIcon } from '@heroicons/react/24/outline'
+import { t } from 'i18next'
+import { useTranslation } from 'react-i18next'
 
 export const GearPage = () => {
+  const { t, i18n } = useTranslation(['gear'])
+  const [searchGoods, setSearchGoods] = useState('')
+  const [maxPrice, setMaxPrice] = useState<number>(Infinity)
+  const [itemAvailability, setItemAvailability] = useState<Record<
+    string,
+    Availability
+  > | null>(null)
+
+  // create a filterarray of goods that match the search string and/or max price
+  const filteredGoods = consumerGoods.filter((goods: ConsumerGoods) => {
+    const matchesSearch = goods.item
+      .toLowerCase()
+      .includes(searchGoods.toLowerCase())
+    const isWithinPriceRange = goods.price.copper <= maxPrice
+
+    return matchesSearch && isWithinPriceRange
+  })
+
+  const filteredServices = regularServices.filter((service: RegularService) => {
+    const matchesSearch = service.service
+      .toLowerCase()
+      .includes(searchGoods.toLowerCase())
+    const isWithinPriceRange = service.price.copper <= maxPrice
+
+    return matchesSearch && isWithinPriceRange
+  })
+
+  const refreshAvailability = () => {
+    const availability: Record<string, Availability> = {}
+    consumerGoods.forEach((item) => {
+      availability[item.item] = availabilityFormat(
+        item.availability,
+      ) as Availability
+    })
+    regularServices.forEach((item) => {
+      availability[item.service] = availabilityFormat(
+        item.availability,
+      ) as Availability
+    })
```

Comment:
This is fine for now but might need to be changed in the future for possible key collisions